source/core/TDCodecVia.cpp

```diff
@@ -1736,6 +1739,15 @@ void TDViaParser::ParseClump(VIClump* viClump, InstructionAllocator* cia)
             if (!_string.EatChar(')'))
                 return LOG_EVENT(kHardDataError, "')' missing");
             state.MarkPerch(&perchName);
+        } else if (instructionNameToken.CompareCStr("DebugPoint") && _isDebuggingDisabled) {
```

Comment:
@shivaprasad-basavaraj Why is debugging opt out instead of opt in? Shouldn't Vireo ignore DebugPoint instructions unless we opt into debugging? There is no benefit to opting into running DebugPoint instructions by default right?

I think instead of `isDebuggingDisabled` the flag should be `debugging` and by default `debugging` is false.

Reply:
@rajsite I went back and forth on it a few times. The reason I made debugging opt out is that the user did choose to generate the DebugPoint instructions so I thought the normal flow in the execution engine should be to respect those instructions and the optional flow would be even if you did generate DebugPoint instructions don't consider them during execution. My thinking was that during a normal flow its the compiler who controls whether debug instructions should be emitted or not and the runtime just runs whatever instructions it has.

Once we have a concept of enabling/disabling debugging in the GWeb editor, I think we would skip generation of DebugPoints altogether.

Reply:
I still think it should be opt-in:
- all it does is add overhead to execution. If a user is running in a debug state they are using the runtime in an environment that does extra work to utilize the behavior of the debug instructions already. In any other context it just adds overhead.
- all else equal, the flag debugging is clearer than isDebuggingDisabled

Reply:
I can even give a somewhat concrete example, if PicoG adopts the new vireo and Gweb with Debug point it'll run slower by default https://github.com/PicoG/PicoVireo/blob/main/platform/rp2040/main.cpp#L220
They get no benefit from it being on by default unless they opt-in and add debugging features to compliment DebugPoint. By default it'll just regress in perf until they out out.

Reply:
I hadn't thought of the case where the user would use new GWeb and new Vireo in conjunction. I have now made the parsing of debugging instructions opt-in.